### PR TITLE
Adding class target to RequiresDynamicCodeAttribute

### PR DESCRIPTION
**Pull request by LakshanF**
#67368 was approved

### CODE REVIEW

**Review by danmoseley — APPROVED** [review]

**Review by tlakollo — APPROVED** [review]
Looks good to me, lets's just wait for CI to come green. Not sure if there is any APICompat that could break stuff in any of the branches

File: src/libraries/System.Runtime/ref/System.Runtime.cs [hunk 1]
```diff
@@ -7368,7 +7368,7 @@ public sealed partial class RequiresAssemblyFilesAttribute : System.Attribute
         public string? Message { get { throw null; } }
         public string? Url { get { throw null; } set { } }
     }
-    [System.AttributeUsageAttribute(System.AttributeTargets.Constructor | System.AttributeTargets.Method, Inherited=false)]
+    [System.AttributeUsageAttribute(System.AttributeTargets.Constructor | System.AttributeTargets.Method | AttributeTargets.Class, Inherited=false)]
```
Comment (eerhardt) [inline src/libraries/System.Runtime/ref/System.Runtime.cs:7371]: I assume you didn't generate this.

```suggestion
    [System.AttributeUsageAttribute(System.AttributeTargets.Constructor | System.AttributeTargets.Method | System.AttributeTargets.Class, Inherited=false)]
```

I wonder if we can even generate System.Runtime anymore...

Comment (eerhardt) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:7371]: Looks like there are steps:

https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/updating-ref-source.md#for-systemruntime

My only concern is that these are probably ordered differently, and so hand-editing this is going to cause more diffs in the future.

**Review by stephentoub — APPROVED** [review]
Other than Eric's comments, LGTM.

File: src/libraries/System.Runtime/ref/System.Runtime.cs (hunk 1 above)
Comment (tlakollo) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:7371]: My major problem with that script is that as far as the last time I executed it didn't understand nullable and it was deleted. This created a big diff file out of which I will nitpick the change I wanted

Comment (eerhardt) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:7371]: The simplest thing here is to see what order the tool generates these flag enums, and then hand edit the file to match it. The more diffs we have from the generated code, the harder this is on everyone else in the future.

Comment (tlakollo) [reply src/libraries/System.Runtime/ref/System.Runtime.cs:7371]: Created an issue to track how can we improve the experience https://github.com/dotnet/runtime/issues/67784

**Review by eerhardt — APPROVED** [review]
LGTM

**Pull request merged**